plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/IcebergRestCatalogModule.java

```diff
@@ -36,5 +42,20 @@ protected void setup(Binder binder)
                 new NoneSecurityModule()));
 
         binder.bind(TrinoCatalogFactory.class).to(TrinoIcebergRestCatalogFactory.class).in(Scopes.SINGLETON);
+        newOptionalBinder(binder, IcebergFileSystemFactory.class).setBinding().to(IcebergRestCatalogFileSystemFactory.class).in(Scopes.SINGLETON);
+
+        IcebergConfig icebergConfig = buildConfigObject(IcebergConfig.class);
+        IcebergRestCatalogConfig restCatalogConfig = buildConfigObject(IcebergRestCatalogConfig.class);
+        if (restCatalogConfig.isVendedCredentialsEnabled() && icebergConfig.isRegisterTableProcedureEnabled()) {
+            throw new TrinoException(NOT_SUPPORTED, "Using the `register_table` procedure with vended credentials is currently not supported");
+        }
+    }
+
+    @AssertTrue(message = "Using the `register_table` procedure with vended credentials is currently not supported")
```

Comment:
Can we put it in the config class even though the properties span two different configs? The register table config is relevant to all catalogs but the vended credential one is only bound for rest catalogs.

Reply:
Ah, no, that's not possible, unless you added the register table config property to the REST catalog config object. This would technically work, as multiple config objects can use the same config name, but I wouldn't do that here, as it's ugly to do just for some validation.

We can remove this method, as the check is done directly above, and `@AssertTrue` is only useful for bean classes which use bean validation.

Reply:
Good point, removed.